src/pages/HomePage/HomePage.tsx

```diff
@@ -0,0 +1,32 @@
+import { useAppSelector } from '../../app/hooks';
+import { CourseList } from '../../components/CourseList';
+import { ErrorNotification } from '../../components/ErrorNotification';
+import { Loader } from '../../components/Loader';
+import { ErrorType } from '../../types/ErrorType';
+
+export const HomePage: React.FC = () => {
+  const {
+    courses,
+    hasError,
+    loaded,
+  } = useAppSelector(state => state.courses);
+
+  return (
+    <>
+      {hasError && (
+        <ErrorNotification
+          error={ErrorType.GET_COURSES}
+        />
+      )}
+
+      {!loaded && <Loader />}
+
+      {courses.length > 0
+        && !hasError
+        && loaded
+        && (
+          <CourseList />
```

Comment:
Here we get **courses** from the useAppSelector hook. In the CourseList component, we also use this hook to get the same list of courses. So you can pass "courses" to the prop in the CourseList component. This will help avoid using the useAppSelector hook twice.

`<CourseList courses={courses} />`

Reply:
Thank you for your recommendation. 😉

But in this case I think it'll better to avoid props. Usually, using of the Redux hook **useAppSelector** is more preferable than passing props to components.

**useAppSelector** allows easy access to the state from the Redux store without passing it through props. This avoids the need to pass the state through multiple levels of components and makes the code more readable and understandable.

In addition, **useAppSelector** automatically subscribes to changes in the Redux state and triggers a re-render of the component upon state changes, which can also simplify the code. 🙌

Reply:
I got it, thanks for the explanation, it makes sense